catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloader.java

```diff
@@ -642,9 +549,45 @@ public ResourceResponse getResourceResponse() {
         return resourceResponse;
     }
 
+    /**
+     * Gets the download status from the reliable resource status.
+     *
+     * @return the DownloadStatus accessed through the ReliableResourceStatus object.
+     */
+    public DownloadStatus getDownloadStatus() {
+        return reliableResourceStatus.getDownloadStatus();
+    }
+
+    /**
+     * Gets the entire ReliableResourceStatus object.
+     *
+     * @return the ReliableResourceStatus object, reliableResourceStatus.
+     */
+    public ReliableResourceStatus getReliableResourceStatus() {
+        return reliableResourceStatus;
+    }
+
+    /**
+     * Get the bytes already downloaded of the reliable resource.
+     *
+     * @return The number of bytes already downloaded for the reliable resource.
+     */
+    public long getReliableResourceByteSize() {
+        return reliableResourceByteSize;
+    }
+
+    /**
+     * Gets the Resource object created during setup.
+     *
+     * @return the Resource object resource.
+     */
+    public Resource getResource() {
+        return resource;
+    }
+
     @VisibleForTesting
-    void setFileOutputStream(FileOutputStream fos) {
-        this.fos = fos;
+    void setFile(File cacheFile) {
```

Comment:
We should look and see if there is a way to eliminate these factory methods marked with `@VisibleForTesting`.